include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,742 @@
+      }
+    }
+  }
+
+  template <size_t _PaddingStride, class _Size>
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct(const _Extents &__extents, _Size __padding_value)
+  {
+    const auto __s = __find_aligned_offset(__padding_value, __extents.extent(_ExtentToPadRank));
+
+    return __subs_type::__construct(__extents, extents<typename _Extents::index_type, _ActualPaddingStride>(__s));
+  }
+
+  template <size_t _PaddingStride, class _IndexType>
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct_other(const _Extents &__extents, _IndexType __padding_stride)
+  {
+    return __subs_type::__construct(__extents, extents<typename _Extents::index_type, _ActualPaddingStride>(__padding_stride));
+  }
+};
+
+template<size_t _ExtentToPadRank, class _Extents, size_t _ActualPaddingStride>
+struct __inner_extents_type<_ExtentToPadRank, _Extents, _ActualPaddingStride, enable_if_t<_Extents::rank() <= size_t(1)>>
+{
+  using __type = _Extents;
+
+  template<size_t _PaddingStride>
+  MDSPAN_INLINE_FUNCTION
+  static constexpr const _Extents &
+  __construct(const _Extents &__extents)
+  {
+    return __extents;
+  }
+
+  template <size_t _PaddingStride, class _Size>
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct(const _Extents &__extents, _Size)
+  {
+    return __extents;
+  }
+
+  template <size_t _PaddingStride, class _IndexType>
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct_other(const _Extents &__extents, _IndexType)
+  {
+    return __extents;
+  }
+};
+
+template <size_t _ExtentToPadRank, class _Extents, typename _Enabled = void>
+struct __unpadded_extent_type_impl
+{
+  using __type = extents<typename _Extents::index_type, _Extents::static_extent(_ExtentToPadRank)>;
+
+  MDSPAN_INLINE_FUNCTION
+  static constexpr auto
+  __construct(const _Extents &__extents)
+  {
+    return __type(__extents.extent(_ExtentToPadRank));
+  }
+};
+
+template <size_t _ExtentToPadRank, class _Extents>
+struct __unpadded_extent_type_impl<_ExtentToPadRank, _Extents, enable_if_t<(_Extents::rank() == 0)>>
+{
+  using __type = extents<typename _Extents::index_type>;
+
+  MDSPAN_INLINE_FUNCTION
+  static constexpr auto
+  __construct([[maybe_unused]] const _Extents &__extents)
+  {
+    return __type{};
+  }
+};
+}
+
+template <size_t padding_stride>
+template <class Extents>
+class layout_left_padded<padding_stride>::mapping {
+public:
+  using extents_type = Extents;
+  using index_type = typename extents_type::index_type;
+  using size_type = typename extents_type::size_type;
+  using rank_type = typename extents_type::rank_type;
+  using layout_type = layout_left_padded<padding_stride>;
+
+#ifndef MDSPAN_INTERNAL_TEST
+private:
+#endif // MDSPAN_INTERNAL_TEST
+
+  static constexpr rank_type __padding_stride_rank = 1;
+  static constexpr rank_type __extent_to_pad_rank = 0;
+
+  static_assert((padding_stride != 0)
+                    || (extents_type::static_extent(__extent_to_pad_rank) == 0)
```

Comment:
Would you consider reformatting so that the `||` goes at the end of the line?  The spacing here looks a bit odd.

Reply:
I personally prefer the operator at the beginning of the line for readability; however you're right that the spacing is weird. I retabbed it so that the || is aligned with the opening parenthesis